### PR TITLE
task: add artifacts

### PR DESCRIPTION
**Pull request by YohannParis**
* Added a an extra steps to downloads all artifacts defined in previous jobs.
* Set an ENV named `ARTIFACT` to be used by `docker-bake.hcl` to use.

### CODE REVIEW

File: .github/workflows/bake-publish.yaml [hunk 1]
```diff
@@ -107,27 +121,28 @@ jobs:
           echo "VERSION=${TAG,,}" >> ${GITHUB_ENV}
           echo "DOCKER_REGISTRY=${REG,,}" >> ${GITHUB_ENV}
           echo "DOCKER_ORG=${ORG,,}" >> ${GITHUB_ENV}
+          echo "ARTIFACT=${{ ARTIFACT_PATH }}${{ inputs.artifact }}" >> ${GITHUB_ENV}
```
Comment (Nithos) [inline .github/workflows/bake-publish.yaml]: do we need a `/` between the ENV and the inputs value?

Comment (Nithos) [reply .github/workflows/bake-publish.yaml]: whats your use within the bake file? is this just meant to be used in the 
```
dockerfile: ${ARTIFCT}
```

Comment (YohannParis) [reply .github/workflows/bake-publish.yaml]: Yes, it's for the bake file to have a path a file as needed, or a folder if someone wants:
```hcl
target "skema-text-reading-base" {
  context = "."
  tags = tag("skema-text-reading", "", "")
  dockerfile = "${ARTIFACT}"
}
```

Comment (Nithos) [reply .github/workflows/bake-publish.yaml]: Ok, so I would try linking to this branch directly in your skema repo and testing this next to see if it works before committing

**Review by Nithos — APPROVED** [review]

**Review by Nithos — CHANGES_REQUESTED** [review]
Need to modify the other publish file as well or remove it

File: .github/workflows/bake-publish.yaml [hunk 2]
```diff
@@ -107,27 +122,28 @@ jobs:
...
-      # 2.4 - Setup QEMU for platform emulation
+      # 2.5 - Setup QEMU for platform emulation
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v2
 
-      # 2.5 - Setup Docker BuildX for multi platform building
+      # 2.6 - Setup Docker BuildX for multi platform building
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
         with:
```
Comment (Nithos) [inline .github/workflows/bake-publish.yaml:142]: please remove this as well, or make a separate PR for this, it should no longer be needed.
It was pinned to a version due to an upgrade that was done earlier this year that caused issues, I believe this has been resolved now as I am not using it in any other version except in ASKEM

Comment (Nithos) [reply .github/workflows/bake-publish.yaml:142]: comment seems off...
I mean the `with:` section that sets the driver version

**Review by Nithos — APPROVED** [review]

File: .github/workflows/bake-publish.yaml (hunk 2 above)
Comment (Nithos) [reply .github/workflows/bake-publish.yaml:142]: Can be done as a separate PR. Feel free to resolve this and merge.

**Pull request merged**